lib/class-wp-rest-image-editor-controller.php

```diff
@@ -101,90 +104,149 @@ public function permission_callback( $request ) {
 	public function apply_edits( $request ) {
 		require_once ABSPATH . 'wp-admin/includes/image.php';
 
-		$params = $request->get_params();
+		$params        = $request->get_params();
+		$attachment_id = $params['media_id'];
 
-		$media_id = $params['media_id'];
+		// This also confirms the attachment is an image.
+		$image_file = wp_get_original_image_path( $attachment_id );
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
 
-		// Get image information.
-		$attachment_info = wp_get_attachment_metadata( $media_id );
-		$media_url       = wp_get_attachment_image_url( $media_id, 'original' );
+		if ( ! $image_meta || ! $image_file ) {
+			$error = __( 'Unable to get meta information for file.', 'gutenberg' );
+			return new WP_Error( 'rest_unknown_attachment', $error, array( 'status' => 404 ) );
+		}
+
+		// Check if we need to do anything.
+		$rotate = 0;
+		$crop   = false;
 
-		if ( ! $attachment_info || ! $media_url ) {
-			return new WP_Error( 'rest_unknown_attachment', __( 'Unable to get meta information for file.', 'gutenberg' ), array( 'status' => 404 ) );
+		if ( ! empty( $params['rotation'] ) ) {
+			// Rotation direction: clockwise vs. counter clockwise.
+			$rotate = 0 - (int) $params['rotation'];
 		}
 
-		$meta = array( 'original_name' => basename( $media_url ) );
+		if ( isset( $params['x'], $params['y'], $params['width'], $params['height'] ) ) {
+			$crop = true;
+		}
 
-		if ( isset( $attachment_info['richimage'] ) ) {
-			$meta = array_merge( $meta, $attachment_info['richimage'] );
+		if ( ! $rotate && ! $crop ) {
+			$error = __( 'The image was not erdited. Edit the image before applying the changes.', 'gutenberg' );
```

Comment:
```suggestion
			$error = __( 'The image was not edited. Edit the image before applying the changes.', 'gutenberg' );
```

Reply:
Ouch, thanks! Would a 400 be appropriate error there? At this point it's an "empty" request, nothing to do.

Reply:
Yeah I think a `400` makes sense.